### PR TITLE
First try at using the stack to find the script that caused a issue i…

### PR DESCRIPTION
**Pull request by sparker256**
…n a function.

Starting attempts at using the stack to find out when a function or callback fails that we know which script it came from so we can put it in quarantine.

### CODE REVIEW

File: src/FloatingWindows/FLWIntegration.cpp [hunk 1]
```diff
@@ -692,15 +695,31 @@ void FindAndQuarantine (lua_State *L)
...
+    {
+        flywithlua::logMsg(logToDevCon,
+               ("FlyWithLua Info: Could not move bad script to " + quarantine_path_name));
+        // LuaIsRunning = false;
+        // Added this break to prevent CTD from flooding the Log.txt file.
+        // break;
+    }
+    XPLMSpeakString("Please reload Your Lua Script files.");
+    // flywithlua::LuaIsRunning = false;
+    // return flywithlua::ReadAllScriptFiles();
```
Comment (sparker256) [inline src/FloatingWindows/FLWIntegration.cpp]: If this is enabled we get a CTD and using gdb here is where it crashes.
`Thread 1 (Thread 0x7ffff7f25a40 (LWP 5244)):
#0  0x00007fff9e4b0853 in propagatemark () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#1  0x00007fff9e4b11e8 in lj_gc_fullgc () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#2  0x00007fff9e49d825 in lua_gc () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#3  0x00007fff9e3e9b88 in flywithlua::ReadScriptFile(char const*) () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#4  0x00007fff9e3f0491 in flywithlua::ReadAllScriptFiles() () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#5  0x00007fff9e3f2201 in flywithlua::FlyWithLuaMenuHandler(void*, void*) () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/FlyWithLua/64/lin.xpl
#6  0x00007fff9ea17f9f in XPLMHandleMenuItemPick(int) () from /home/bill/X-Plane_11.31r1_light/Resources/plugins/XPLM_64.so
#7  0x0000000001129182 in XPPHandleMenu(int) ()
#8  0x000000000083c7a3 in gui_menubar_widget::end_drag(float, float, gui_menubar_widget::end_mode) ()
#9  0x00000000008283ea in gui_event_dispatcher::event_mouse_up(click_event_info_t const&) ()
#10 0x000000000099f1ad in xplane_window::mouse_up(void*, int, int, int) ()
#11 0x00000000014d1538 in WIN_event_loop(int) ()
#12 0x00000000014d16c7 in WIN_run_app ()
#13 0x000000000049d65f in main ()`

Comment (nevkontakte) [reply src/FloatingWindows/FLWIntegration.cpp]: I think the problem is that  ReadAllScriptFiles() destroys Lua context. This means that you can't do this in a function, which was originally called by Lua: once that function returns, control is passed back to Lua is the *destroyed* context. From that point, anything can happen, including a crash.

A workaround is detect the bad script name and throw an exception. The exception will unwind Lua stack, and then we'll catch it at the top, where we originally passed control to Lua. we can create a custom exception class, which will carry the name of the bad script. At that point it should be safe to quarantine the script, re-initialize Lua environment and try again.

Comment (sparker256) [reply src/FloatingWindows/FLWIntegration.cpp]: I am using a exception that was already happening and then trying to reload the remaining scripts again. I need a little help in how to create a exception here if possible.

File: src/FloatingWindows/FLWIntegration.cpp [hunk 2]
```diff
@@ -201,10 +200,9 @@ void LuaSetOnDrawCallback(sol::light<FloatingWindow> wnd, CallbackProvider const
 
         auto on_draw = on_draw_provider();
         if (!on_draw) {
-            std::ostringstream oss_window_id;
-            oss_window_id << "FlyWithLua Info: This XPLMWindowID " << window << " is for LuaCreateFloatingWindow";
-            flywithlua::logMsg(logToAll, oss_window_id.str());
             flywithlua::panic("FlyWithLua Error: invalid or nil window builder passed to float_wnd_set_ondraw");
+            // Need to find a way for this to work when we are using sol.
+            // FindAndQuarantine (L);
```
Comment (sparker256) [inline src/FloatingWindows/FLWIntegration.cpp]: Need to find how to get to the stack info when we are using sol to finish up finding all the errors.

**Pull request merged**